### PR TITLE
fix: make directive not throw error in server context

### LINKED ISSUE #101: Angular Universal SSR issue due to `window` being present
Since there is no window context on the server, all the `window` references makes the server side rendering fail. 

I'd suggest to use some browser checking (universal has one) before using `window`.

### PR DESCRIPTION
**Pull request by kevinphelps**
This fixes #101.

### CODE REVIEW

File: src/scroll-listener.ts [hunk 1]
```diff
@@ -17,7 +17,9 @@ export function sampleObservable(obs: Observable<any>, scheduler?: any) {
 // Typical, there will only be one observable per application
 export const getScrollListener = (scrollTarget): Observable<any> => {
     if (!scrollTarget || typeof scrollTarget.addEventListener !== 'function') {
-        console.warn('`addEventListener` on ' + scrollTarget + ' (scrollTarget) is not a function. Skipping this target');
+        if (typeof window !== 'undefined') {
+            console.warn('`addEventListener` on ' + scrollTarget + ' (scrollTarget) is not a function. Skipping this target');
+        }
```
Comment (tjoskar) [inline src/scroll-listener.ts]: Why do we need to wrap `console.warn` with this if statement?

Comment (kevinphelps) [inline src/scroll-listener.ts]: So that the warning doesn't pollure node server logs.

Comment (kevinphelps) [inline src/scroll-listener.ts]: So that the warning doesn't pollute node server logs.

**Comment by tjoskar** [comment]
Looks awesome, thanks @kevinphelps! I'm just curious about the if-statement above but I will merge this PR.

File: src/scroll-listener.ts (hunk 1 above)
Comment (tjoskar) [inline src/scroll-listener.ts]: ✅

**Comment by tjoskar** [comment]
Thanks @kevinphelps 🎉

**Comment by kevinphelps** [comment]
You're welcome! Thanks for merging. :)

**Pull request merged**